lib/vagrant_cloud/data.rb

```diff
@@ -0,0 +1,265 @@
+module VagrantCloud
+  # Generic data class
+  class Data
+    # Custom nil class
```

Comment:
Since this kinda seems a little odd at first glace, I wonder if it'd be worth a couple of sentences in comments for why we have our own nil class?